### PR TITLE
Docs/fix recently broken links

### LINKED ISSUE #14968: Broken link on enterprise documentation pages
Most (all?) of the enterprise feature documentation pages have a bit at the top about how it is an enterprise feature...

ent-box

The link to `enterprise feature matrix` (https://developer.hashicorp.com/docs/enterprise#consul-enterprise-feature-availability) is broken.

### PR DESCRIPTION
**Pull request by jkirschner-hashicorp**
Closes #14968.

Also fixes other issues introduced by the movement of content that used to be on the main `/api-docs` page.

Best reviewed by commit.

### CODE REVIEW

File: website/content/docs/enterprise/admin-partitions.mdx [hunk 1]
```diff
@@ -11,7 +11,7 @@ description: >-
   This feature requires version 1.11.0+ of
   HashiCorp Cloud Platform (HCP) or self-managed Consul Enterprise.
   Refer to the{' '}
-  <a href="/docs/enterprise#consul-enterprise-feature-availability">enterprise feature matrix</a>
+  <a href="/consul/docs/enterprise#consul-enterprise-feature-availability">enterprise feature matrix</a>
```
Comment (jkirschner-hashicorp) [inline website/content/docs/enterprise/admin-partitions.mdx:14]: For some reason, the auto-translation to the new URLs (prefixing them with `/consul`) doesn't seem to apply to these links specified in manual `<a>` elements.

File: agent/keyring.go [hunk 2]
```diff
@@ -116,20 +116,25 @@ LOAD:
 	// Only perform the following checks if there was an encrypt_key
 	// provided in the configuration.
 	if rtConfig.EncryptKey != "" {
+		// Ignored gossip encryption key:
```
Comment (jkirschner-hashicorp) [inline agent/keyring.go:119]: This shouldn't be here...

File: website/content/api-docs/acl/auth-methods.mdx [hunk 3]
```diff
@@ -28,7 +28,7 @@ The table below shows this endpoint's support for
 [blocking queries](/api-docs/features/blocking),
 [consistency modes](/api-docs/features/consistency),
 [agent caching](/api-docs/features/caching), and
-[required ACLs](/api-docs#authentication).
+[required ACLs](/api-docs/api-structure#authentication).
```
Comment (jkirschner-hashicorp) [inline website/content/api-docs/acl/auth-methods.mdx:31]: This is basically the same change in a **TON** of places

**Review by david-yu — APPROVED** [review]

File: website/content/api-docs/kv.mdx [hunk 4]
```diff
@@ -33,7 +33,7 @@ If the [`recurse`](#recurse) or [`keys`](#keys) query parameters are `true`,
 this endpoint will return an array of keys. In this case,
 the HTTP response includes the `X-Consul-Results-Filtered-By-ACLs: true` header
 if the response array excludes results due to ACL policy configuration.
-Refer to the [HTTP API documentation](/api-docs#results-filtered-by-acls) for more information.
+Refer to the [HTTP API documentation](/api-docs/api-structure#results-filtered-by-acls) for more information.
```
Comment (jkirschner-hashicorp) [inline website/content/api-docs/kv.mdx:36]: This works

**Pull request merged**